crates/prql_compiler/src/semantic/context.rs

```diff
@@ -9,7 +9,7 @@ use crate::ir::pl::*;
 
 /// Context of the pipeline.
 #[derive(Default, Serialize, Deserialize, Clone)]
-pub struct Context {
+pub struct ResolvedQuery {
```

Comment:
Yeah, name `Context` is too generic.

Although this does not include just a query, but all declarations: functions, types, modules.

Maybe we could call this `Environment`?

Reply:
Environment isn't much better ... I think I actually prefer Context ... I think the main thing that bothered me was that lower_to_iq took just a `Context` ... but this can be resolved by introducing a new `ResolvedQuery` type like in the commit I just pushed ... please take a look.

Reply:
That's the key insight: lowering takes a set of resolved declarations, picks the one pointed to by `main_ident` and lowers that to `RelationalQuery`.

The `ResolvedQuery` struct you've created now also contains the `main_ident`, so more appropriate name would be `LoweringRequest`? But this is weird.

What about renaming `Context` to `ResolvedDeclarations`? A bit long...

Reply:
I'd say the lowering requires a main_ident it cannot take place when there isn't one.

Reply:
I'd agree, yes